Framework/Algorithms/src/ConjoinWorkspaces.cpp

```diff
@@ -208,5 +247,16 @@ bool ConjoinWorkspaces::processGroups() {
   return retval;
 }
 
+void ConjoinWorkspaces::setYUnitAndLabel(API::MatrixWorkspace &ws) const {
+  const std::string yLabel = getPropertyValue("YAXisLabel");
+  const std::string yUnit = getPropertyValue("YAxisUnit");
+
+  if (!yUnit.empty())
```

Comment:
Is it worth putting a comment here saying the unit must be moved before label as changing the unit resets the label?